river/Cursor.zig

```diff
@@ -487,6 +498,42 @@ fn handleRequestSetCursor(
     }
 }
 
+pub fn hide(self: *Self) void {
+    if (self.pressed_count > 0) return;
+    self.hidden = true;
+    self.wlr_cursor.setImage(null, 0, 0, 0, 0, 0, 0);
+    self.image = .unknown;
+    self.seat.wlr_seat.pointerNotifyClearFocus();
+    self.hide_cursor_timer.timerUpdate(0) catch {
+        log.err("failed to update cursor hide timeout", .{});
+    };
+}
+
+pub fn unhide(self: *Self) void {
+    self.hide_cursor_timer.timerUpdate(server.config.cursor_hide_timeout) catch {
+        log.err("failed to update cursor hide timeout", .{});
+    };
+    if (!self.hidden) return;
+    self.hidden = false;
+    if (self.surfaceAt()) |result| {
+        if (self.seat.wlr_seat.pointer_state.focused_surface != result.surface) {
+            switch (result.parent) {
+                .view => {
+                    self.seat.wlr_seat.pointerNotifyEnter(result.surface, result.sx, result.sy);
+                },
+                .layer_surface => {},
+                .xwayland_unmanaged => assert(build_options.xwayland),
+            }
+        }
+    }
```

Comment:
This could be replaced with a call to `self.updateState()`, which would also be a bit more correct I believe.